docs/rules.md

```diff
@@ -0,0 +1,46 @@
+# Rules
+
+*Rules* in FMN consist of several “sub-rules”: One [*Tracking Rule*](#tracking-rules), and
+potentially many [*Filter Rules*](#filter-rules) and [*Destination Rules*](#destination-rules)
+(grouped into [*Generation Rules*](#generation-rules)). The *Message Consumer* looks at all messages
+transported over the bus and if any match a *Rule*, *Notifications* will be sent to the respective
+*Notification Sender*, one instance of each is responsible for actually sending out emails and chat
+messages over IRC or Matrix, respectively.
+
+## Tracking Rules
+
+Each *Rule* has exactly one *Tracking Rule* which specifies what messages should be tracked, e.g.
+messages that concern:
+
+* Artifacts owned by the user or a group the user is member of,
+* specific named artifacts,
+* the user themselves, or
+* followed users.
+
+If a *Tracking Rule* matches, its [*Generation Rules*](#generation-rules) will be consulted for
+further processing.
+
+## Generation Rules
+
+Each *Rule* contains one or more *Generation Rules* which group together zero or more [*Filter
+Rules*](#filter-rules) and one or more [*Destination Rules*](#destination-rules). If no [*Filter
+Rules*](#filter-rules) exist, or all of them match, *Notification Objects* will be created for each
+[*Destination Rule*](#destination-rules) which are processed by the respective *Sender*.
+
+This lets users e.g. specify that messages of a lower severity should be sent via email, while
+higher severities should let FMN ping the user via IRC or Matrix.
+
+## Filter Rules
+
+*Filter Rules* further restrict if a message should be matched by a *Rule* and notifications should
+be sent. Users can configure filters for these criteria:
+
+* The name of the application sending a message.
+* The severity of the message.
+* If a message was the cause by an action of the user.
+* If the message topic matches a certain glob pattern.
+
+## Destination Rules
+
+A *Destination Rule* contains information about how a user should be notified if a *Rule* matches,
+e.g. via email, IRC or Matrix.
```

Comment:
I would add that the available destinations are retrieved from the user's account and can be set in Noggin.